x-pack/plugins/ml/server/lib/alerts/register_anomaly_detection_alert_type.ts

```diff
@@ -137,18 +184,36 @@ export function registerAnomalyDetectionAlertType({
     minimumLicenseRequired: MINIMUM_FULL_LICENSE,
     isExportable: true,
     doesSetRecoveryContext: true,
-    async executor({ services, params, spaceId }) {
+    executor: async ({
+      services,
+      params,
+      spaceId,
+    }: ExecutorOptions<MlAnomalyDetectionAlertParams>) => {
       const fakeRequest = {} as KibanaRequest;
       const { execute } = mlSharedServices.alertingServiceProvider(
         services.savedObjectsClient,
         fakeRequest
       );
+
+      const { alertsClient } = services;
+
       const executionResult = await execute(params, spaceId);
 
       if (executionResult && !executionResult.isHealthy) {
-        const alertInstanceName = executionResult.name;
-        const alertInstance = services.alertFactory.create(alertInstanceName);
-        alertInstance.scheduleActions(ANOMALY_SCORE_MATCH_GROUP_ID, executionResult.context);
+        alertsClient?.report({
+          id: executionResult.name,
+          actionGroup: ANOMALY_SCORE_MATCH_GROUP_ID,
+          context: executionResult.context,
+          payload: expandFlattenedAlert({
+            [ALERT_URL]: executionResult.context.anomalyExplorerUrl,
+            [ALERT_ANOMALY_DETECTION_JOB_ID]: executionResult.context.jobIds[0],
+            [ALERT_ANOMALY_SCORE]: executionResult.context.score,
+            [ALERT_ANOMALY_IS_INTERIM]: executionResult.context.isInterim,
+            [ALERT_ANOMALY_TIMESTAMP]: executionResult.context.timestamp,
+            [ALERT_TOP_RECORDS]: executionResult.context.topRecords,
+            [ALERT_TOP_INFLUENCERS]: executionResult.context.topInfluencers,
+          }),
+        });
       }
 
       // Set context for recovered alerts
```

Comment:
should switch to the alerts client for reporting recovered alerts as well

Reply:
done in https://github.com/elastic/kibana/pull/166349/commits/bf07373a3a8f3f56061e8b19a6b68770b60b692d